### PR TITLE
 BGDIINF_SB-2349: the - is being used for time queries

### PR DESCRIPTION
**Pull request by rebert**

### CODE REVIEW

**Comment by rebert** [comment]
The commit 4b2ce33ebbfd5749d3832be966e48782e8096e83 is escaping the -, which makes that queries on timeperiods do not work anymore:
http://mf-chsdi3.prod.bgdi.ch/rest/services/all/SearchServer?type=featuresearch&searchText=1990-2010&features=ch.swisstopo.fixpunkte-lfp1%2Cch.swisstopo.lubis-luftbilder_schwarzweiss%2Cch.swisstopo.lubis-luftbilder_farbe&timeEnabled=false%2Ctrue%2Ctrue

Which again makes the e2e-tests fail:
```
→>$ make test TEST_NAMES=tests.test_service_search.test_search_features.TestSearchFeatures.test_search_features_time_interval ENV=dev.env
pipenv run dotenv -f /home/ltrea/infra-e2e-tests/config/dev.env run pipenv run nose2 \
-c nose2_test.cfg \
-t /home/ltrea/infra-e2e-tests \
-v \
 \
 \
tests.test_service_search.test_search_features.TestSearchFeatures.test_search_features_time_interval
Courtesy Notice: Pipenv found itself running within a virtual environment, so it will automatically use that environment, instead of creating its own for any project.
test_search_features_time_interval (tests.test_service_search.test_search_features.TestSearchFeatures) ... ERROR

======================================================================
ERROR: test_search_features_time_interval (tests.test_service_search.test_search_features.TestSearchFeatures)
----------------------------------------------------------------------
Traceback (most recent call last):
  File "/home/ltrea/infra-e2e-tests/tests/test_service_search/test_search_features.py", line 233, in test_search_features_time_interval
    self.assertEqual(response.json()['results'][0]['attrs']['origin'], 'feature')
IndexError: list index out of range

-------------------- >> begin captured logging << --------------------
curlified: DEBUG: 
	 $: curl -X GET -H "If-None-Match: "0be49c12-fe4d-4a2f-8991-c75ba79dee75"" -H "user-agent: python-requests/e2e-tests" https://sys-api3.dev.bgdi.ch/rest/services/ech/SearchServer?type=featuresearch&searchText=1993034+1990-2010&features=ch.swisstopo.fixpunkte-lfp1%2Cch.swisstopo.lubis-luftbilder_schwarzweiss%2Cch.swisstopo.lubis-luftbilder_farbe&timeEnabled=false%2Ctrue%2Ctrue
urllib3.connectionpool: DEBUG: Starting new HTTPS connection (1): sys-api3.dev.bgdi.ch:443
urllib3.connectionpool: DEBUG: https://sys-api3.dev.bgdi.ch:443 "GET /rest/services/ech/SearchServer?type=featuresearch&searchText=1993034+1990-2010&features=ch.swisstopo.fixpunkte-lfp1%2Cch.swisstopo.lubis-luftbilder_schwarzweiss%2Cch.swisstopo.lubis-luftbilder_farbe&timeEnabled=false%2Ctrue%2Ctrue HTTP/1.1" 200 35
curlified: DEBUG: Response headers for get https://sys-api3.dev.bgdi.ch/rest/services/ech/SearchServer?type=featuresearch&searchText=1993034+1990-2010&features=ch.swisstopo.fixpunkte-lfp1%2Cch.swisstopo.lubis-luftbilder_schwarzweiss%2Cch.swisstopo.lubis-luftbilder_farbe&timeEnabled=false%2Ctrue%2Ctrue:
	Content-Type:application/json
	Content-Length:35
	Connection:keep-alive
	Server:gunicorn
	Date:Wed, 04 May 2022 12:13:59 GMT
	Content-Encoding:gzip
	Cache-Control:public, max-age=600
	Access-Control-Allow-Methods:GET, HEAD, OPTIONS
	Access-Control-Allow-Headers:*
	Access-Control-Allow-Origin:*
	X-Cache:Miss from cloudfront
	Via:1.1 6b0e09b8a7d995016df1513b4b11c17e.cloudfront.net (CloudFront)
	X-Amz-Cf-Pop:ZRH50-C1
	X-Amz-Cf-Id:tvOxPCkyUZs5uGFzziCjKcvafzcu8wHRJb3LXbbyCfmZSchV-SpQ6w==
curlified: DEBUG: Response for get https://sys-api3.dev.bgdi.ch/rest/services/ech/SearchServer?type=featuresearch&searchText=1993034+1990-2010&features=ch.swisstopo.fixpunkte-lfp1%2Cch.swisstopo.lubis-luftbilder_schwarzweiss%2Cch.swisstopo.lubis-luftbilder_farbe&timeEnabled=false%2Ctrue%2Ctrue:
{'results': []}
curlified: DEBUG: 
	 $: curl -X GET -H "If-None-Match: "2d86a151-ab70-440a-ab6a-76e1f9d5bbe9"" -H "user-agent: python-requests/e2e-tests" https://sys-api3.dev.bgdi.ch/rest/services/ech/SearchServer?type=featuresearch&searchText=1990-2010&features=ch.swisstopo.fixpunkte-lfp1%2Cch.swisstopo.lubis-luftbilder_schwarzweiss%2Cch.swisstopo.lubis-luftbilder_farbe&timeEnabled=false%2Ctrue%2Ctrue
urllib3.connectionpool: DEBUG: https://sys-api3.dev.bgdi.ch:443 "GET /rest/services/ech/SearchServer?type=featuresearch&searchText=1990-2010&features=ch.swisstopo.fixpunkte-lfp1%2Cch.swisstopo.lubis-luftbilder_schwarzweiss%2Cch.swisstopo.lubis-luftbilder_farbe&timeEnabled=false%2Ctrue%2Ctrue HTTP/1.1" 200 35
curlified: DEBUG: Response headers for get https://sys-api3.dev.bgdi.ch/rest/services/ech/SearchServer?type=featuresearch&searchText=1990-2010&features=ch.swisstopo.fixpunkte-lfp1%2Cch.swisstopo.lubis-luftbilder_schwarzweiss%2Cch.swisstopo.lubis-luftbilder_farbe&timeEnabled=false%2Ctrue%2Ctrue:
	Content-Type:application/json
	Content-Length:35
	Connection:keep-alive
	Server:gunicorn
	Date:Wed, 04 May 2022 12:13:59 GMT
	Content-Encoding:gzip
	Cache-Control:public, max-age=600
	Access-Control-Allow-Methods:GET, HEAD, OPTIONS
	Access-Control-Allow-Headers:*
	Access-Control-Allow-Origin:*
	X-Cache:Miss from cloudfront
	Via:1.1 6b0e09b8a7d995016df1513b4b11c17e.cloudfront.net (CloudFront)
	X-Amz-Cf-Pop:ZRH50-C1
	X-Amz-Cf-Id:IQi2nL6zd0C5mX-xLOEy5otvgYrMiQotBGDGRBOTZZFyDT5QMEs3Ng==
curlified: DEBUG: Response for get https://sys-api3.dev.bgdi.ch/rest/services/ech/SearchServer?type=featuresearch&searchText=1990-2010&features=ch.swisstopo.fixpunkte-lfp1%2Cch.swisstopo.lubis-luftbilder_schwarzweiss%2Cch.swisstopo.lubis-luftbilder_farbe&timeEnabled=false%2Ctrue%2Ctrue:
{'results': []}
utils.base: INFO: Tear Down test...
```

**Review by ltshb — COMMENTED** [review]
A unittest to make sure that `-` does not get escaped would be good. Also adding a bit more detailed in git commit message and jira key 😉 
But otherwise very good catch 👍🏼

File: tests/unit_tests/test_helpers.py [hunk 1]
```diff
@@ -36,6 +36,11 @@ def test_format_search_text(self):
         result = format_search_text(testinput_str)
         self.assertEqual(result, '%+\\&\\/\\(\\)\\=?\\!ueaeoeea\\$\\@i\\$')
 
+    def test_minus_not_being_escaped(self):
+        testinput_str = '1940-2022'
+        result = format_search_text(testinput_str)
+        self.assertEqual(result, '1940-2022')
```
Comment (ltshb) [inline tests/unit_tests/test_helpers.py:42]: 👍🏼

**Review by ltshb — APPROVED** [review]
Great thanks for the job

**Pull request merged**